tests/unit/parts/test_setup_assets.py

```diff
@@ -176,6 +178,23 @@ def test_kernel_missing(yaml_data, new_dir):
     assert not kernel_path.is_file()
 
 
+@pytest.mark.parametrize(
+    "source,destination,copied",
+    [
+        (Path("foo"), Path("bar"), True),
+        (Path("foo"), Path("foo"), False),
+    ],
+)
+def test_copy_files(source, destination, copied, new_dir, mocker):
```

Comment:
you are missing a test or scenario for "samefile" but different contents

Reply:
also, legacy has link_or_copy (maybe already in new snapcraft) which might solve some issues around existing files

Reply:
The "samefile" test should be inode-based so different content must mean different files, but added a case with same name and different content.

Reply:
`link_or_copy` was not originally used because it dragged a large infrastructure to deal with only a couple of files, but now that more files are copied we can consider bringing it in a refactoring PR.